RATIONALE.md

```diff
@@ -0,0 +1,69 @@
+# Rationale for api designs
+
+## Get multiple traces: /api/v2/traceMulti?traceIds=id1,id2
+Get by multiple trace IDs was added in [/api/v2](zipkin2-api.yaml) to support
+more efficient requests when trace IDs are found by means besides the query api.
+
+Here are some use cases:
+* out-of-band aggregations, such histograms with attached representative IDs
+* trace comparison views
+* decoupled indexing and retrieval (Ex lucene over a blob store)
+* 3rd party apis that offer stable pagination (via a cursor over trace IDs).
+
+There were several design concerns to address when integrating another feature
+into an existing api root. Most of these dealt with how to inform consumers that
+the feature exists.
+
+### Why not just use http cache control on /trace/{traceId}?
+It is a reasonable question why not just introduce a caching layer on trace ID,
+and have callers make successive requests. There are a number of problems with
+this. First, it would only help if cache-control headers were sent, but they
+have never been sent. This is because results of a trace are unstable until
+complete. Also, only use cases that revisit the same trace IDs are aided with
+caching, for example trace comparison views. It will always cost more to prime
+the cache with multiple requests vs one.
+
+Finally, bear in mind not all use cases revisit the same trace IDs, or even
+support http cache control. For example, if someone were paging over all traces
+to perform an aggregation, those reads only happen once. In this case, the value
+of efficiently getting a bucket of results is foreground.
+
+### Why not just have this an internal detail of storage?
+As time goes by, we've accumulated more use cases of multi-get. When there are
+more than 3 integrations, we typically consider normalizing a feature.
+
+### Why /traceMulti instead of /trace/id,id2
+A separate endpoint than /trace/ allows us to help consumers, such as the UI,
+know the difference between unsupported (404) and empty (200 with empty list
+response). /trace/{traceId}, on the other hand, returns 404 on not found. This
+brings up a larger problems that a result of multiple traces is a list of
+traces, which is a different data type than a list of spans.
+
+### Why /traceMulti instead of /traces?traceIds=id1,id2
+When search is disabled, `/trace/{traceId}` is still mounted, eventhough
+`/traces` is not. Multi-get is an optimization of `/trace/{traceId}`, so needs
+to be mounted when search-like mechanisms are disabled. It is still tempting to
+re-use the same endpoint, but there is another problem. A new parameter will not
+fail queries in most implementations.
+
+For example, calling `/traces?traceIds=id1,id2` in an existing service is most
+likely to return as if there was no `traceIds` parameter at all, resulting in
+random results returned.
+
+### Why /traceMulti
```

Comment:
Not a native but `traceMany` sound a little better. Feel free to completely ignore this comment.

Reply:
Agreed on use of many